src/algorithms/banditpam.cpp

```diff
@@ -13,588 +13,516 @@
 
-    medoidIndicesFinal = medoidIndices;
-    labels = assignments;
+  medoidIndicesBuild = medoidIndices;
+  arma::urowvec assignments(data.n_cols);
+  if (nMedoids > 1) {
+    BanditPAM::swap(data, distMat, &medoidIndices, &medoidMatrix, &assignments);
   }
 
-  arma::frowvec BanditPAM::buildSigma(
-          const arma::fmat &data,
-          std::optional<std::reference_wrapper<const arma::fmat>> distMat,
-          const arma::frowvec &bestDistances,
-          const bool useAbsolute) {
-    size_t N = data.n_cols;
-    // Temporarily increase the batch size
-    // for precise estimation of the standard deviation
-    int originalBatchSize = batchSize;
-    batchSize = std::min(1000, static_cast<int>(N/4));
-    arma::uvec referencePoints;
-    // TODO(@motiwari): Make this wraparound properly as
-    //  last batch_size elements are dropped
-    if (usePerm) {
-      if ((permutationIdx + batchSize - 1) >= N) {
-        permutationIdx = 0;
-      }
-      // inclusive of both indices
-      referencePoints = permutation.subvec(
-              permutationIdx,
-              permutationIdx + batchSize - 1);
-      permutationIdx += batchSize;
-    } else {
-      referencePoints = arma::randperm(N, batchSize);
+  medoidIndicesFinal = medoidIndices;
+  labels = assignments;
+}
+
+arma::frowvec BanditPAM::buildSigma(
+  const arma::fmat &data,
+  std::optional<std::reference_wrapper<const arma::fmat>> distMat,
+  const arma::frowvec &bestDistances, const bool useAbsolute) {
+  size_t N = data.n_cols;
+  // Temporarily increase the batch size
+  // for precise estimation of the standard deviation
+  int originalBatchSize = batchSize;
+  batchSize = std::min(1000, static_cast<int>(N / 4));
+  arma::uvec referencePoints;
+  // TODO(@motiwari): Make this wraparound properly as
+  //  last batch_size elements are dropped
+  if (usePerm) {
+    if ((permutationIdx + batchSize - 1) >= N) {
+      permutationIdx = 0;
     }
+    // inclusive of both indices
+    referencePoints =
+      permutation.subvec(permutationIdx, permutationIdx + batchSize - 1);
+    permutationIdx += batchSize;
+  } else {
+    referencePoints = arma::randperm(N, batchSize);
+  }
 
-    arma::fvec sample(batchSize);
-    arma::frowvec updated_sigma(N);
-    #pragma omp parallel for if (this->parallelize)
-    for (size_t i = 0; i < N; i++) {
-      for (size_t j = 0; j < batchSize; j++) {
-        // 0 for MISC
-        float cost =
-                KMedoids::cachedLoss(data, distMat, i,
-                                     referencePoints(j), 0);
-        if (useAbsolute) {
-          sample(j) = cost;
-        } else {
-          sample(j) = cost < bestDistances(referencePoints(j))
-                        ? cost : bestDistances(referencePoints(j));
-          sample(j) -= bestDistances(referencePoints(j));
-        }
+  arma::fvec sample(batchSize);
+  arma::frowvec updated_sigma(N);
+#pragma omp parallel for if (this->parallelize)
```

Comment:
indentation! for some reason, the new style change seems to think `#pragma` as a function?